apps/api/src/billing/services/trial-validation/trial-validation.service.ts

```diff
@@ -6,16 +6,18 @@ import { singleton } from "tsyringe";
 
 import { UserWalletOutput } from "@src/billing/repositories";
 import { AUDITOR, TRIAL_ATTRIBUTE, TRIAL_REGISTERED_ATTRIBUTE } from "@src/deployment/config/provider.config";
-import { getAddressDeployments } from "@src/services/external/apiNodeService";
+import { DeploymentReaderService } from "@src/deployment/services/deployment-reader/deployment-reader.service";
 import type { UserOutput } from "@src/user/repositories";
 
 const TRIAL_DEPLOYMENT_LIMIT = 5;
 
 @singleton()
 export class TrialValidationService {
+  constructor(private readonly deploymentReaderService: DeploymentReaderService) {}
+
   async validateTrialLimit(decoded: EncodeObject, userWallet: UserWalletOutput) {
     if (userWallet.isTrialing && decoded.typeUrl === "/akash.deployment.v1beta3.MsgCreateDeployment") {
-      const deployments = await getAddressDeployments(userWallet.address, 0, 1, false, {});
+      const deployments = await this.deploymentReaderService.listByOwner(userWallet.address, 0, 1, false, {});
```

Comment:
Please change the interface of this method to an object. Having number/boolean args is impossible to understand.
Compare to: 

```suggestion
      const deployments = await this.deploymentReaderService.listByOwner({ address: userWallet.address, limit: 1 });
```